03/app.js

```diff
@@ -11,4 +11,30 @@ const buttonSettings = {
         color: '#444'
     },
     text: 'Click me!',
-}
\ No newline at end of file
+}
+
+const newButton  = document.createElement('button');
+
+
+for (settingName in buttonSettings) {
+    typeOfSetting = buttonSettings[settingName];
+    //atributes + styles
+    if(typeof typeOfSetting==='object'){
+        //attributes
+        if (settingName==='attr') {
+            for (property in typeOfSetting) {
+                newButton[property]= `${typeOfSetting[property]}`;
+            }
+        }
+        //styles
+        for (property in typeOfSetting) {
+            newButton.style[property]= `${typeOfSetting[property]}`;
+        }
+    //text
+    } else {
+        newButton.innerText = typeOfSetting;
+    }
+}
```

Comment:
Może czytelniej będzie jak zrobisz to rozwiązanie oddzielnie dla każdej właściwości tj. np. `for (const attr in buttonSettings.attr) {` :)